src/main/kotlin/oop/State/SemaphoreState.kt

```diff
@@ -0,0 +1,12 @@
+package oop.State
+
+sealed class SemaphoreState {
+
+  class GreenSemaphore : SemaphoreState()
```

Comment:
https://www.youtube.com/watch?v=cnOA7HdNUR4&index=2&list=PLKxa4AIfm4pUaVhRKxYErhAvjU9xqwZOb you can see here